tests/analysis/test_distribution_transformer.py

```diff
@@ -1,18 +1,18 @@
-import random
-
 import probably.pgcl as pgcl
 import pytest
-from pytest import raises
 
-import prodigy.analysis
+from prodigy.analysis.analyzer import compute_discrete_distribution
 from prodigy.analysis.config import ForwardAnalysisConfig
 from prodigy.distribution.fast_generating_function import ProdigyPGF
 from prodigy.distribution.generating_function import SympyPGF
+from prodigy.distribution.symengine_distribution import SymenginePGF
 
 
-@pytest.mark.parametrize('engine,factory',
-                         [(ForwardAnalysisConfig.Engine.SYMPY, SympyPGF),
-                          (ForwardAnalysisConfig.Engine.GINAC, ProdigyPGF)])
+@pytest.mark.parametrize('engine,factory', [
```

Comment:
ideally, get the available engines and factories from the forwardAnalysisConfig and then automatically parametrize the test in all engine, factory combinations.